### DISCUSSION TITLE
Formattazione custom sul numero fattura

### DISCUSSION THREAD

**Discussion by Alexxia5** [fattureincloud/api#425] [Q&A] [answered]
Ciao, avrei necessità di esporre il numero fattura nel formato  2024000545 concatenando quindi l'anno + 6 cifre per il numero con gli zeri a sinistra.
Inoltre la prima fattura che dovrà essere emessa dovrà partire da un numero specifico.
E' possibile farlo?

Grazie mille
Alex

**Comment by emajo** [comment, marked as answer]
Ciao @Alexxia5, è possibile fare ciò che dici creando una fattura e settando manualmente il numero a 2024000545, creando poi altre fatture e lasciando la numerazione incrementare automaticamente la fattura successiva sarà numerata 2024000546 e così via.

Grazie,
Emanuele.

**Reply by Alexxia5** [reply]
Ciao Emanuele, 
Scusami stavo vedendo il risultato della numerazione e funziona come mi hai detto. 
Ho solo un problema perchè sulla fatture dopo la numerazione concatena di nuovo l'anno: 2024000546/2024. 
C'è un modo per eliminare la parte dopo 2024000546 ?
grazie

**Reply by emajo** [reply]
Ciao @Alexxia5, sta impostando il sezionale (campo 'numeration') con il valore 2024? In quel caso basta omettere il campo

**Reply by Alexxia5** [reply]
no, il valore del numeration non l'ho settato. 
 IssuedDocument invoice = new IssuedDocument()
                .type(IssuedDocumentType.INVOICE)
                .entity(entity)
                .date(dettaglioFattura.getDataFattura())
              //  .number(2024000545) // se si omette il numero verrà incrementato automaticamente
                //.numeration(dettaglioFattura.getNumeration())
                .subject(dettaglioFattura.getSubject()) // non visibile in fattura
                .visibleSubject(dettaglioFattura.getVisibleSubject()) // visibile in fattura
                .currency(new Currency().id("EUR"))
                .language(new Language()
                        .code("it")
                        .name("italiano")
                )

il valore dell'anno non l'ho settato da nessuna parte. (a parte il campo data fattura e number che lo contengono )

**Reply by emajo** [reply]
Ciao @Alexxia5, nell'xml dovrebbe vedersi il numero corretto, quel /2024 credo sia solo mostrato nel pdf, e dovrebbe anche essere personalizzabile a livello di template.

Grazie,
Emanuele.

**Reply by valmoz** [reply]
Buongiono @Alexxia5 possiamo chiudere la discussione?

Grazie
Mauro

**Reply by Alexxia5** [reply]
Si va bene.
Vi ringrazio molto del supporto

**Comment by Alexxia5** [comment]
Grazie mille.
Il cambio anno poi vedo come gestirlo.
Intanto grazie.
:)
